src/amber/router/segments/variable_segment.cr

```diff
@@ -19,5 +26,22 @@ module Amber::Router
     def parameter : String
       segment[1..-1]
     end
+
+    private def match_integer?(segment : String) : Bool
+      segment.each_byte do |byte|
+        return false unless 48 <= byte <= 57
```

Comment:
I'm rusty on unicode, what will this do if a unicode multibyte string is provided?

Reply:
I only made this for ascii I will have to check unicode but I am unfamiliar with it.